File: backends/dotnet/src/Zappy.Application/Accounts/ListSessions.cs
using Zappy.Domain;

namespace Zappy.Application;

public sealed class ListSessions(ISessionRepository sessions, IClock clock)
{
    public async Task<IReadOnlyList<Session>> Execute(string customerId, CancellationToken cancellationToken) =>
        await sessions.OpenOfCustomer(customerId, clock.Now, cancellationToken);
}

File: backends/kotlin/zappy-application/src/main/kotlin/nl/zappymart/application/accounts/ListSessions.kt
package nl.zappymart.application.accounts

import nl.zappymart.application.ports.Clock
import nl.zappymart.application.ports.SessionRepository
import nl.zappymart.domain.accounts.Session

class ListSessions(
    private val sessions: SessionRepository,
    private val clock: Clock,
) {

    fun execute(customerId: String): List<Session> = sessions.findOpenForCustomer(customerId, clock.moment())
}
